src/main/java/frc/robot/FieldConstants.java

```diff
@@ -14,12 +19,36 @@
  * reduce verbosity.
  */
 public final class FieldConstants {
+    // Field size, copied from AprilTag layout
+    public static final double FIELD_LENGTH = 16.54175;
+    public static final double FIELD_WIDTH = 8.0137;    
+    
     // safeZone x limit on blue side, cannot stow in this zone
     public static final double BAD_ZONE_X_BLUE = 2.0;
     // safeZone x limit on red side, cannot stow in this zone
-    public static final double BAD_ZONE_X_RED = 14.6;
+    public static final double BAD_ZONE_X_RED = FIELD_LENGTH - BAD_ZONE_X_BLUE;
```

Comment:
why not just use flipX() here :)

Reply:
You can't rely or use the Alliance during the initialization.